AdyenDropIn/Models/PaymentMethodsConfiguration.swift

```diff
@@ -27,7 +27,18 @@ public extension DropInComponent {
         
         /// The client key that corresponds to the webservice user you will use for initiating the payment.
         /// See https://docs.adyen.com/user-management/client-side-authentication for more information.
-        public var clientKey: String
+        public var clientKey: String {
+            didSet {
+                environment.clientKey = clientKey
+            }
+        }
+
+        /// The API environment to use.
+        public var environment: Environment {
+            didSet {
+                environment.clientKey = clientKey
```

Comment:
this could be confusing if you generated a new instance of Environment with `clientKey` and assigned it to `PaymentMethodsConfiguration`.

I would prefer to make it required and constant:

```swift 
  public let clientKey: String
  public let environment: Environment
  
  public init(clientKey: String, environment: Environment) {
      self.clientKey = clientKey            
      self.environment = environment            
      self.environment.clientKey = clientKey
  }
```

Reply:
Merchants are not supposed to create environment with a `clientKey`, the fact that environment has the client key is sort of hidden, he client key set in the `clientKey` property is the one that should be used.

Reply:
All I am saying is that it is not obvious to the merchant that `Environment` have `clientKey` in it..
I would rather treat those two entities as unrelated constants

Reply:
i'm also not very on board about this. why is it not a constant? since we are injecting it at construction time, and it is not supposed to change

Reply:
made them immutable